localstack/services/ec2/exceptions.py

```diff
@@ -54,3 +56,21 @@ def __init__(self):
             code="InvalidLaunchTemplateId.VersionNotFound",
             message="Could not find launch template version",
         )
+
+
+class InvalidSecurityGroupNotFound(CommonServiceException):
+    def __init__(self, name: Any):
+        super().__init__(
+            "InvalidGroup.NotFound",
+            f"The security group '{name}' does not exist",
+        )
+
+
+class InvalidGroupIdMalformed(CommonServiceException):
+    def __init__(
+        self,
+    ):
+        super().__init__(
+            "InvalidGroupId.Malformed",
+            "The security group ID '' is malformed",
```

Comment:
the security group ID is omitted: `''`

Reply:
Hey @viren-nadkarni ,

When I invoke the ```describe-security-group-rules``` command with an empty group-id value on AWS it returns an error message ```InvalidGroupId.Malformed```

```bash
aws ec2 describe-security-group-rules --filters "Name=group-id,Values=['']"
An error occurred (InvalidGroupId.Malformed) when calling the DescribeSecurityGroupRules operation: The security group ID '' is malformed
```
And when I use the same on localstack,it also returns an error, but with a different message:
```bash
awslocal ec2 describe-security-group-rules --filters "Name=group-id,Values=['']"
An error occurred (InvalidGroup.NotFound) when calling the DescribeSecurityGroupRules operation: The security group 'No security groups found matching the filters provided.' does not exist
```

It appears that Moto is returning InvalidGroup.NotFound for both cases when the group-id value is empty and when the security group doesn't exist. 
Could you please confirm if this behaviour in Moto is intentional?
If that's the case, I'm willing to make the necessary adjustments.

Reply:
Looks like a bug in Moto. We should make it as close to AWS behaviour as possible